src/lib/components/PublishButton/SubmitAndPublishReviewButton.js

```diff
@@ -0,0 +1,129 @@
+// This file is part of React-Invenio-Deposit
+// Copyright (C) 2023 CERN.
+//
+// React-Invenio-Deposit is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+import React, { Component } from "react";
+import { i18next } from "@translations/i18next";
+import _get from "lodash/get";
+import { connect } from "react-redux";
+import { Button } from "semantic-ui-react";
+import { connect as connectFormik } from "formik";
+import {
+  DepositFormSubmitActions,
+  DepositFormSubmitContext,
+} from "../../DepositFormSubmitContext";
+import { SubmitAndPublishReviewModal } from "./modals";
+import { DepositStatus } from "../../state/reducers/deposit";
+import _omit from "lodash/omit";
+import PropTypes from "prop-types";
+
+class SubmitAndPublishReviewButtonComponent extends Component {
+  state = { isConfirmModalOpen: false };
+  static contextType = DepositFormSubmitContext;
+
+  openConfirmModal = () => this.setState({ isConfirmModalOpen: true });
+
+  closeConfirmModal = () => this.setState({ isConfirmModalOpen: false });
+
+  handleSubmitReview = () => {
+    const { formik } = this.props;
+    const { handleSubmit } = formik;
+    const { setSubmitContext } = this.context;
+
+    setSubmitContext(DepositFormSubmitActions.SUBMIT_REVIEW, {
+      directPublish: true,
```

Comment:
To me, this param is only when we want to force it.
We could submit a review without the param, and still have direct publish (it we have rights to do so).
The code will also probably be much easier.